File: 프로그래머스/1/42748. K번째수/K번째수.py
# 문제 분석
# 1. 슬라이싱 i - 1 ..< j
# 2. 오름차순 정렬
# 3. k - 1 번째 숫자 return

def solution(array, commands):
    answer = []
    
    for com in commands:
        i, j, k = com[0], com[1], com[2]
        slicedArr = array[i - 1 : j]
        sortedArr = sorted(slicedArr)
        answer.append(sortedArr[k - 1])
    
    return answer

File: 프로그래머스/1/42748. K번째수/K번째수.swift
// 자르고 [i - 1 ... j - 1]
// 정렬하고
// k번째 [k - 1]

import Foundation

func solution(_ array:[Int], _ commands:[[Int]]) -> [Int] {
    
    var result = [Int]()

    for command in commands {
        let (left, right, k_idx) = (command[0], command[1], command[2])
        result.append(Array(array[left - 1 ... right - 1]).sorted()[k_idx - 1])
    }
    
    return result
}